labs/lab3/chapter3.md

```diff
@@ -222,37 +215,61 @@ $ curl localhost:8080 #note we indicated the port to use in the run command abov
   * `--network=container:<alias>` to link to the wordpress container
 ```bash
 $ ls -lZd ~/workspace/mysql
-$ sudo podman run -d --network=container:wordpress -v ~/workspace/logs:/var/log/mariadb:z -v ~/workspace/mysql:/var/lib/mysql:z -v ~/workspace/run:/run/mariadb:z -e DBUSER=user -e DBPASS=mypassword -e DBNAME=mydb --name mariadb mariadb
+$ sudo podman run -d --network=container:wordpress -v ~/workspace/mysql:/var/lib/mysql:z -e DBUSER=user -e DBPASS=mypassword -e DBNAME=mydb --name mariadb mariadb
 ```
 Note: See the difference in SELinux context after running w/ a volume & :z.
 ```bash
-$ ls -lZd ~/workspace/mysql
+$ ls -lZ ~/workspace/mysql
 $ sudo podman exec mariadb ps aux
 ```
 
 Check volume directory ownership inside the container
 ```bash
-$ sudo podman exec mariadb stat --format="%U" /var/log/mariadb/mariadb.log
+$ sudo podman exec mariadb stat --format="%U" /var/lib/mysql/mysql
 ```
 
 Now we can check out how the database is doing
 ```bash
 $ sudo podman logs mariadb
 $ sudo podman ps
 $ sudo podman exec mariadb curl localhost:3306
+$ sudo podman exec mariadb mysql -u user --password=mypassword -e 'show databases'
 $ curl localhost:3306 #as you can see the db is not generally visible
 $ curl -L http://localhost:8080 #and now wp is happier!
 ```
 
 You may also load the Wordpress application in a browser to test its full functionality @ `http://<YOUR AWS VM PUBLIC DNS NAME HERE>:8080`.
 
+## Deploy a Container Registry
+
+Let's deploy a simple registry to store our images.
+
+Inspect the Dockerfile that has been prepared.
+```bash
+$ cd ~/summit-2018-container-lab/labs/lab3/
+$ cat registry/Dockerfile
+```
+
+Build & run the registry
+```bash
+$ sudo podman build -t registry registry/
+$ mkdir ~/workspace/registry
+$ sudo podman run --name registry -p 5000 -v ~/workspace/registry:/var/lib/registry:z -d registry
+```
+
+Confirm the registry is running.
+```bash
+$ sudo podman ps
+```
+
```

Comment:
so.. I am not sure having persistent storage for the registry is a good idea. I mean, it is a good idea, but, in the lab, it makes it harder to reset. I think. Basically, I am mixed and am curious on other people's opinion.

Reply:
i guess i'm good either way. fwiw, any new image build that's pushed to the registry should overwrite a prior one even w/ persistent storage.

is that what you mean by "harder to reset"?

Reply:
going to go ahead and merge this in so i can complete my review without merge conflicts.  we need to keep this item open though.

Reply:
i say we leave it the same as what we had before. no persistent storage for the registry

Reply:
@whitel it won't affect our ability to recreate the container...  but ill go ahead and remove it.

Reply:
@whitel @scollier @dustymabe removed the persistent storage